src/app/store/author/authors.actions.ts

```diff
@@ -0,0 +1,48 @@
+import { createAction, props } from "@ngrx/store";
+import { AuthorsConstant } from "./authors.constants";
+
+import { Author } from "@app/shared/models/author.model";
+
+export const requestAllAuthors = createAction(
+  AuthorsConstant.REQUEST_ALL_AUTHORS
+);
+
+export const requestAllAuthorsSuccess = createAction(
+  AuthorsConstant.REQUEST_ALL_AUTHORS_SUCCESS,
+  props<{ authors: Author[] }>()
+);
+
+export const requestAllAuthorsFail = createAction(
+  AuthorsConstant.REQUEST_ALL_AUTHORS_FAIL,
+  props<{ error: string }>()
+);
+
+export const requestAuthorsById = createAction(
+  AuthorsConstant.REQUEST_AUTHORS_BY_ID,
+  props<{ authors: string[] }>()
```

Comment:
should be `{ author: Author }`, right?

Reply:
@TamasViczian  as previous comment, I rethinked the getAuthorByID function too, based on the same reason ( allAuthors containes my authors exist, and I thought it is better to search in this allAuthors in the store and return back with the authors)
It is authors: string[] because the authors saved by id-s in an Array on the server, so I use this string array. I can imagine scenarios when it is better to request the Authors one by one but for this enumeration isn't it enought to use the array from the store? As you can see ( in the previous task) I am able to implement the getAuthorById through API calls too. If it is mandatory I'll change it back.